diff --git a/Makefile b/Makefile
index 5f4e414..a18a074 100644
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,10 @@
-CFLAGS=-pipe -mfloat-abi=hard -mcpu=arm1176jzf-s -fomit-frame-pointer -mabi=aapcs-linux -mtune=arm1176jzf-s -mfpu=vfp -Wno-psabi -mno-apcs-stack-check -g -mstructure-size-boundary=32 -mno-sched-prolog
+CFLAGS=-pipe -mfloat-abi=hard -mcpu=cortex-a8 -fomit-frame-pointer -mabi=aapcs-linux -mtune=cortex-a8 -mfpu=neon-vfpv4 -Wno-psabi -mno-apcs-stack-check -g -mstructure-size-boundary=32 -mno-sched-prolog
 CFLAGS+=-std=c++0x -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -DTARGET_POSIX -DTARGET_LINUX -fPIC -DPIC -D_REENTRANT -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CMAKE_CONFIG -D__VIDEOCORE4__ -U_FORTIFY_SOURCE -Wall -DHAVE_OMXLIB -DUSE_EXTERNAL_FFMPEG  -DHAVE_LIBAVCODEC_AVCODEC_H -DHAVE_LIBAVUTIL_OPT_H -DHAVE_LIBAVUTIL_MEM_H -DHAVE_LIBAVUTIL_AVUTIL_H -DHAVE_LIBAVFORMAT_AVFORMAT_H -DHAVE_LIBAVFILTER_AVFILTER_H -DHAVE_LIBSWRESAMPLE_SWRESAMPLE_H -DOMX -DOMX_SKIP64BIT -ftree-vectorize -DUSE_EXTERNAL_OMX -DTARGET_RASPBERRY_PI -DUSE_EXTERNAL_LIBBCM_HOST
 
-LDFLAGS=-L$(SDKSTAGE)/opt/vc/lib/
-LDFLAGS+=-L./ -Lffmpeg_compiled/usr/local/lib/ -lc -lbrcmGLESv2 -lbrcmEGL -lbcm_host -lopenmaxil -lfreetype -lz -lasound
+LDFLAGS=-L$(SNAPCRAFT_STAGE)/usr/lib/
+LDFLAGS+=-L./ -Lffmpeg_compiled/usr/local/lib/
 
-INCLUDES+=-I./ -Ilinux -Iffmpeg_compiled/usr/local/include/ -I /usr/include/dbus-1.0 -I /usr/lib/arm-linux-gnueabihf/dbus-1.0/include -I/usr/include/freetype2 -isystem$(SDKSTAGE)/opt/vc/include -isystem$(SDKSTAGE)/opt/vc/include/interface/vcos/pthreads
+INCLUDES+=-I./ -Ilinux -Iffmpeg_compiled/usr/local/include/ -I /usr/include/dbus-1.0 -I /usr/lib/arm-linux-gnueabihf/dbus-1.0/include -I/usr/include/freetype2 -I $(SNAPCRAFT_STAGE)/usr/include -I $(SNAPCRAFT_STAGE)/usr/include/interface/vcos/pthreads
 
 DIST ?= omxplayer-dist
 STRIP ?= strip
@@ -53,7 +53,7 @@ version:
 	bash gen_version.sh > version.h 
 
 omxplayer.bin: version $(OBJS)
-	$(CXX) $(LDFLAGS) -o omxplayer.bin $(OBJS) -lvchiq_arm -lvchostif -lvcos -ldbus-1 -lrt -lpthread -lavutil -lavcodec -lavformat -lswscale -lswresample -lpcre
+	$(CXX) $(LDFLAGS) -o omxplayer.bin $(OBJS) -lc -lbrcmGLESv2 -lbrcmEGL -lbcm_host -lopenmaxil -lfreetype -lz -lasound -lvchiq_arm -lvchostif -lvcos -ldbus-1 -lrt -lpthread -lavutil -lavcodec -lavformat -lswscale -lswresample -lpcre
 	$(STRIP) omxplayer.bin
 
 help.h: README.md Makefile
@@ -94,7 +94,7 @@ dist: omxplayer.bin omxplayer.1
 	cd $(DIST); tar -czf ../$(DIST).tgz *
 
 install:
-	cp -r $(DIST)/* /
+	cp -r $(DIST)/* $(DESTDIR)/
 
 uninstall:
 	rm -rf /usr/bin/omxplayer
diff --git a/Makefile.ffmpeg b/Makefile.ffmpeg
index 0ed4419..b93930e 100644
--- a/Makefile.ffmpeg
+++ b/Makefile.ffmpeg
@@ -20,11 +20,11 @@ configure:
 	CFLAGS="$(CFLAGS) ${INCLUDES}" \
 	LDFLAGS="" \
   ./configure \
-			--extra-cflags="-mfpu=vfp -mfloat-abi=hard -mno-apcs-stack-check -mstructure-size-boundary=32 -mno-sched-prolog" \
+			--extra-cflags="-mfpu=neon-vfpv4 -mfloat-abi=hard -mno-apcs-stack-check -mstructure-size-boundary=32 -mno-sched-prolog" \
 			--enable-shared \
 			--disable-static \
 			--arch=arm \
-			--cpu=arm1176jzf-s \
+			--cpu=cortex-a8 \
 			--target-os=linux \
 			--disable-hwaccels \
 			--enable-parsers \
@@ -46,7 +46,7 @@ configure:
 			--enable-pthreads \
 			--enable-pic \
 			--disable-armv5te \
-			--disable-neon \
+			--enable-neon \
 			--enable-armv6t2 \
 			--enable-armv6 \
 			--enable-hardcoded-tables \
diff --git a/omxplayer.cpp b/omxplayer.cpp
index 0d4ab5e..fb238a4 100644
--- a/omxplayer.cpp
+++ b/omxplayer.cpp
@@ -83,8 +83,8 @@ bool              m_osd                 = true;
 bool              m_no_keys             = false;
 std::string       m_external_subtitles_path;
 bool              m_has_external_subtitles = false;
-std::string       m_font_path           = "/usr/share/fonts/truetype/freefont/FreeSans.ttf";
-std::string       m_italic_font_path    = "/usr/share/fonts/truetype/freefont/FreeSansOblique.ttf";
+std::string       m_font_path           = "/snap/rpi-iptv-player/current/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf";
+std::string       m_italic_font_path    = "/snap/rpi-iptv-player/current/usr/share/fonts/truetype/dejavu/DejaVuSans-Oblique.ttf";
 std::string       m_dbus_name           = "org.mpris.MediaPlayer2.omxplayer";
 bool              m_asked_for_font      = false;
 bool              m_asked_for_italic_font = false;
